dso/dependabot.py

```diff
@@ -0,0 +1,187 @@
+    details = []
+
+    clogger.info(f'scanning {len(repos)} repos')
+    for repo in repos:
+        github_cfg = ccc.github.github_cfg_for_hostname(
+            cfg_factory=ci.util.ctx().cfg_factory(),
+            host_name=host,
+        )
+        clogger.info(f'{i} / {len(repos)}')
+        dependabot_found = is_dependabot_in_component_repo(
+            repo=repo,
+            github_cfg=github_cfg,
+            github=host
+        )
+        repo_report = dso.model.DependabotCoverageReportRepo(
+            dependabot=dependabot_found,
+            repo=repo
+        )
+        details.append(repo_report)
+        i += 1
+
+    return dso.model.DependabotCoverageReport(
+        coverage=0,
+        details=details,
+        github=host,
+    )
+
+
+def _get_github_api(host):
+    github_cfg = ccc.github.github_cfg_for_hostname(
+        cfg_factory=ci.util.ctx().cfg_factory(),
+        host_name=host,
+    )
+    return ccc.github.github_api(github_cfg)
+
+
+def _get_scoped_repos_for_org(
+    org: str,
+) -> typing.List[str]:
+    clogger = dso.util.component_logger(__name__)
+    repos: typing.List[str] = []
+
+    clogger.info(f'org specified, scanning {org}')
+    host, org = org.split('/')
+    github_api = _get_github_api(host)
+    github_org = github_api.organization(org)
+    for repo in github_org.repositories():
+        repos.append(repo.full_name)
+
+    return repos[:3]
+
+
+def _generate_reporting_tables(
+    report: dso.model.DependabotCoverageReport,
+    tablefmt: str,
+):
+    # monkeypatch: disable html escaping
+    tabulate.htmlescape = lambda x: x
+
+    tables = []
+    table_data = (
+        (
+            rr.repo,
+            rr.dependabot,
+        ) for rr in report.details
+    )
+    tables.append(tabulate.tabulate(
+        headers=('Component', 'Dependabot'),
+        tabular_data=table_data,
+        tablefmt=tablefmt,
+        colalign=('left', 'center'),
+    ))
+
+    table_data = ((report.github, f'{report.coverage}%'),)
+    tables.append(tabulate.tabulate(
+        headers=('Overall Coverage', 'Percentage'),
+        tabular_data=table_data,
+        tablefmt=tablefmt,
+        colalign=('left', 'center'),
+    ))
+    return tables
+
+
+def _print_report(
+    report: dso.model.DependabotCoverageReport,
+    outfile_path: str,
+):
+    tables = _generate_reporting_tables(report=report, tablefmt='simple')
+    with open(outfile_path, 'a') as f:
+        now = datetime.datetime.now()
+        f.write(f'{now.strftime("%d/%m/%Y %H:%M:%S")}\n')
+        for t in tables:
+            f.write(f'{t}\n\n')
+            print(f'\n{t}')
+        f.write('\n===================================================================\n\n')
+    print('\n')
+
+
+def dependabot_coverage(
+    org: str,
+    outfile_path: str,
+):
+    repos = _get_scoped_repos_for_org(org=org)
+    host, _ = org.split('/')
+    report = _scan_repos_for_dependabot(repos=repos, host=host)
```

Comment:
so you first create repository-objects (`_get_scoped_repos_for_org'), to then discard those objects again and only use their full-names, which you then use to create repository.-objects _again_? this sounds a bit cumbersome to me (also, it uses two requests for each repository, where we in fact only would need one :-/